app/Console/Commands/DeepCleanDatabase.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Utils\DatabaseCleanupUtils;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\DB;
+
+class DeepCleanDatabase extends Command
+{
+    /**
+     * The name and signature of the console command.
+     */
+    protected $signature = 'db:deep-clean';
```

Comment:
What do you think about just naming this `db:clean`?

Reply:
Originally I made this "deep clean" to contrast with the shallower "mark for potential deletion" approach we discussed for #2468. Thinking about it more though, perhaps it makes sense to remove the builds and the affected shared records all in one command, rather than splitting it in two. In that case, this command could simply be "clean" (no need to distinguish it with "deep").

Reply:
Would it make sense to have one artisan command which runs two separate tasks here?  That way, there's a single command for anyone running things manually, while still maintaining the ability to run the jobs separately inside CDash.

Reply:
Yup, that's what I'm envisioning as well. We can revisit this point once I have something more concrete to share for the next iteration of #2468.